### DISCUSSION TITLE
Faster conversion from numpy to cupy

### DISCUSSION THREAD

**Discussion by saishashank85** [xarray-contrib/cupy-xarray#43] [Show and tell]
def num_to_cud(k):
    k = torch.tensor(k, device='cuda')
    torch.cuda.synchronize()
    k = cp.asarray(k.contiguous())
    cp.cuda.runtime.deviceSynchronize()
    return k

**Comment by jacobtomlinson** [comment]
I don't think adding `pytorch` as a dependency here is a good idea. I'm curious why this is faster and whether you did any measurements of the differences
